src/Datadog.Trace/Tracer.cs

```diff
@@ -361,6 +366,130 @@ private static IStatsd CreateDogStatsdClient(TracerSettings settings, string ser
             return new Statsd(statsdUdp, new RandomGenerator(), new StopWatchFactory(), prefix: string.Empty, constantTags);
         }
 
+        private static void StartStandaloneAgentProcessesWhenConfigured()
+        {
+            try
+            {
+                var datadogConfigPath = Environment.GetEnvironmentVariable(ConfigurationKeys.AgentConfigPath);
+                string traceProcessArgs = null;
+                string dogStatsDArgs = null;
+
+                if (!string.IsNullOrWhiteSpace(datadogConfigPath))
+                {
+                    traceProcessArgs = $"--config \"{datadogConfigPath}\"";
+                }
+
+                _traceAgentMonitor = StartProcessWithKeepAlive(ConfigurationKeys.AgentPath, traceProcessArgs);
+                _dogStatsDMonitor = StartProcessWithKeepAlive(ConfigurationKeys.DogStatsDPath, dogStatsDArgs);
+            }
+            catch (Exception ex)
+            {
+                DatadogLogging.RegisterStartupLog(log => log.Error(ex, "Error when attempting to start standalone agent processes."));
+            }
+        }
+
+        private static bool ProgramIsRunning(string fullPath)
+        {
+            if (string.IsNullOrWhiteSpace(fullPath))
+            {
+                return false;
+            }
+
+            var fileName = Path.GetFileNameWithoutExtension(fullPath).ToLower();
+            var processesByName = Process.GetProcessesByName(fileName);
+
+            if (processesByName.Length > 0)
+            {
+                // We enforce a unique enough naming within contexts where we would use sub-processes
+                return true;
+            }
+
+            return false;
+        }
+
+        private static Task StartProcessWithKeepAlive(string processPathVariable, string args)
```

Comment:
Should all the logic to start a process and keep it alive be in a different class? Seems like it shouldn't be a `Tracer` concern.